### PR TITLE
HW JV OOP Advanced

### PR DESCRIPTION
**Pull request by vrudkovsky**

### CODE REVIEW

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final double radius;
+    private final String color;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:5]:
```suggestion
```
`color `should be in Figure class

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    public Circle(double radius, String color) {
+        this.radius = radius;
+        this.color = color;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:9]:
```suggestion
        super(color);
        this.radius = radius;
```

File: src/main/java/core/basesyntax/Circle.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public double getArea() {
+        return Math.PI * radius * radius;
+    }
+
+    @Override
+    public String getColor() {
+        return color;
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:24]:
```suggestion

```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static final String[] colors = {"red", "orange", "yellow", "green", "blue", "purple"};
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: create `public enum Color`

File: src/main/java/core/basesyntax/Figure.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculator {
+    private String color;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Figure.java:4]:
```suggestion
    protected String color;
```
add one-arg constructor and setter

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+    public static String getRandomColor() {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]: **Don’t use static methods in your solution**
Static methods are in general a bad practice. Let’s better create an instance of a class which method you want to call.

https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#dont-use-static-methods-in-your-solution

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+                return new IsoscelesTrapezoid(random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1, ColorSupplier.getRandomColor());
+            default:
+                return null;
+        }
+    }
+
+    public static Figure getDefaultFigure() {
+        return new Circle(10, "white");
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/FigureSupplier.java:31]: Do not use magic numbers. Create constants

File: src/main/java/core/basesyntax/Square.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        this.color = color;
+    }
+
+    public double getSide() {
+        return side;
+    }
+
+    @Override
+    public double getArea() {
+        return side * side;
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/Square.java:18]: You can use `Math.pow(side, 2)` here

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**